src/store/index.js

```diff
@@ -1,9 +1,17 @@
 import { createStore } from 'vuex';
 
 export default createStore({
-  state: {},
+  state: {
+    // app_modal
+    isModalActive: false,
+  },
   getters: {},
-  mutations: {},
+  mutations: {
+    // app modal
+    toggleAppModal: (state) => {
```

Comment:
mutations should be constants, 
https://vuex.vuejs.org/guide/mutations.html#using-constants-for-mutation-types
Also, following the logic that we will have multiple modals in different pages, the way this is currently setup they would all appear at the same time on toggling, since all three derive their visibility from the same state

You'll need to revise this logic